backend/clubs/views.py

```diff
@@ -451,6 +456,53 @@ def filter_queryset(self, request, queryset, view):
         return new_queryset
 
 
+class ClubFairViewSet(viewsets.ModelViewSet):
+    """
+    list:
+    Return a list of ongoing and upcoming club fairs.
+    """
+
+    http_method_names = ["get", "post"]
+    serializer_class = ClubFairSerializer
+    permission_classes = [ClubFairPermission | IsSuperuser]
+
+    @action(detail=True, methods=["post"])
+    def register(self, request, *args, **kwargs):
+        """
+        Register a club for this club fair.
+        Pass in a "club" string parameter with the club code
+        and a "status" parameter that is true to register the club, or false to unregister.
+        """
+        fair = self.get_object()
+
+        if not request.user.is_authenticated:
+            raise PermissionDenied
+
+        club = get_object_or_404(Club, code=request.data.get("club"))
+
+        # get register/unregister action status
+        status = request.data.get("status")
+        if isinstance(status, str):
+            status = status.strip().lower() == "true"
+        elif not isinstance(status, bool):
+            status = True
+
+        # check if user can actually register club
+        mship = find_membership_helper(request.user, club)
+        if mship is not None and mship.role <= Membership.ROLE_OFFICER or request.user.is_superuser:
```

Comment:
Small stylistic comment:` if mship and ... `? not sure what it's like in the rest of the file

Reply:
If the officer is an owner, then the value will be `0`, which will make this if statement fail instead of passing. This is why I explicitly check for None instead of all falsey values.